browser-switch/src/test/java/com/braintreepayments/browserswitch/BrowserSwitchClientTest.java

```diff
@@ -347,7 +556,38 @@ public void deliverResult_whenActivityIsNotABrowserSwitchListener_throwsIllegalA
         exceptionRule.expectMessage("Activity must implement BrowserSwitchListener.");
 
         sut = BrowserSwitchClient.newInstance(browserSwitchConfig, activityFinder, persistentStore, returnUrlScheme);
-        sut.deliverResult(activity);
+        sut.deliverResult(plainActivity);
+    }
+
+    @Test
+    public void convenience_deliverResultWithActivityListener_forwardsInvocationToPrimaryDeliverResultMethod() {
+        sut = spy(BrowserSwitchClient.newInstance(browserSwitchConfig, activityFinder, persistentStore, returnUrlScheme));
+        doNothing().when(sut).deliverResult(any(FragmentActivity.class), any(BrowserSwitchListener.class));
+
+        sut.deliverResult(activityAndListener);
+        verify(sut).deliverResult(activityAndListener, activityAndListener);
+    }
+
+    @Test
+    public void convenience_deliverResultWithFragmentAndExplicitListener_forwardsInvocationToPrimaryDeliverResultMethod() {
+        when(plainFragment.getActivity()).thenReturn(plainActivity);
+
+        sut = spy(BrowserSwitchClient.newInstance(browserSwitchConfig, activityFinder, persistentStore, returnUrlScheme));
+        doNothing().when(sut).deliverResult(any(FragmentActivity.class), any(BrowserSwitchListener.class));
+
+        sut.deliverResult(plainFragment, browserSwitchListener);
+        verify(sut).deliverResult(plainActivity, browserSwitchListener);
+    }
+
+    @Test
+    public void convenience_deliverResultWithFragmentListener_forwardsInvocationToPrimaryDeliverResultMethod() {
+        when(fragmentAndListener.getActivity()).thenReturn(plainActivity);
+
+        sut = spy(BrowserSwitchClient.newInstance(browserSwitchConfig, activityFinder, persistentStore, returnUrlScheme));
+        doNothing().when(sut).deliverResult(any(FragmentActivity.class), any(BrowserSwitchListener.class));
+
+        sut.deliverResult(fragmentAndListener);
+        verify(sut).deliverResult(plainActivity, fragmentAndListener);
     }
```

Comment:
There's a lot in this test file. Can we organize it into sections by the methods being tested?

It looks like Android Studio supports labeled `region` comments that will code fold.

Ex:

```
//region deliverResult tests

    @Test 
    public void test_Method() { ... }

//endregion
```

Reply:
Added regions 👍